File: code/ex228.py
class Solution:
    def summaryRanges(self, nums):
        """
        :type nums: List[int]
        :rtype: List[str]
        """
        
        # Special case: Blank list
        if nums == []:
            return []


        r = []

        i = 1
        start = 0
        while i < len(nums):
            if nums[i] == nums[i - 1] + 1:
                i += 1
            else:
                if i - start >=2:
                    # If numbers before are continuous
                    r.append(f"{nums[start]}->{nums[i - 1]}")
                else:
                    # If number before is not continuous
                    r.append(f"{nums[start]}")
                

                start = i
                i += 1


        # Add the last item to the result list
        if start == len(nums) - 1:
            r.append(f"{nums[start]}")
        else:
            r.append(f"{nums[start]}->{nums[i - 1]}")

        return r

sol = Solution()
print(sol.summaryRanges([]))

File: code/ex228.cpp
class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        int n = nums.size();    if (n == 0)    return {};
        vector<string> res;
        int prev = 0;
        for (int i = 0; i < n; i++) {
            if (i == 0)    prev = nums[i];
            else {
                if ((long)nums[i] - nums[i - 1] > 1l) {
                    res.push_back(nums[i - 1] - prev == 0 ? to_string(nums[i - 1]) : to_string(prev) + "->" + to_string(nums[i - 1]));
                    prev = nums[i];
                }
            }
        }
        res.push_back(nums[n - 1] - prev == 0 ? to_string(nums[n - 1]) : to_string(prev) + "->" + to_string(nums[n - 1]));
        return res;
    }
};